py3status/modules/external_script.py

```diff
@@ -83,9 +86,14 @@ def external_script(self):
             output_text = ''
 
         response['full_text'] = self.py3.safe_format(
-            self.format, {'output': output_text})
+            self.format, {'output': output_text, 'count_lines': len(output_lines)})
         return response
 
+    def on_click(self, event):
+        button = event["button"]
+        if button != self.button_refresh:
+            self.py3.notify_user(self.output)
```

Comment:
This hardcode the notification with whole output. We want `self.format_notification` for this so users can specify if they want whole output, first line output, or number of lines in an output... just like `format`. It looks like you should make another placeholder to display whole output.

Based on current code, `{output}` is technically `{line_1}` right now because if we try `script_path = 'cal'`, we will get first line only.. instead of whole `cal` in `{output}`.

Reply:
The way I see it... We have a choice. Implement this PR where it will do things wrong and will do things wrong for long time or implement a PR that does things right in first place and not having to worry about it for long time.

For instance, if an user drops in... asking us how to turn off the notification because he simply want to refresh his script without dealing with notifications, can it be turned off right now?

If not, then we have to fix it eventually. That's why I want to fix this right in first place for the notifications. The ideal behavior is to leave notifications off by default too.

Reply:
I don't think not being able to format notification message is necessarily wrong, current behavior shows first line of the output on the status bar, this change allows you to additionally see the rest of the lines on click. 

As for disabling notifications, that's why initially I implemented it with two buttons, button_refresh and button_show_notification. Should I return to that approach?

Reply:
Ask @cyrinux to help you with `format_notification`. It won't be a problem... The problem is that `{output}` placeholder is printing one line... so I assume we have to find a different name that explains `{output}`, but not `{output}`... eg `{full_output}`... but `{output}` is a good name to use.

Reply:
@maximbaz Okay, the solution is make `format_notification = None`, but we won't document `{output_full}` placeholder so only you and few others will know about it... The rest can use `{output}` (aka 1line) or `{line}` (aka count_lines).

If/when we break this module someday, we could do it at that time.

So you just have to implement `format_notification`. Pass `self.py3.safe_form...` object to `self.py3.notifiy_user()`` instead of `self.output` string.

Reply:
Done, now it is possible to set `self.format_notification = None` to disable notification. I intentionally implemented as least as possible functionality to keep this PR focused on its original goal.

Reply:
It is named `output` because it is a proper name, and a short one as you seem to prefer - the fact that in another place `output` currently only prints the first line is the problem of that code.